v2/emailpassword/troubleshooting/cors-issues.mdx

```diff
@@ -0,0 +1,219 @@
+  })  
+);
+```
+
+<br/>
+
+### `[Error] Origin [http://localhost:3000](http://localhost:3000/) is not allowed by Access-Control-Allow-Origin. Status code: 204`
+
+This happens when the CORS origin field is incorrect. Another varient of this error is when the OPTIONS API returns a 404 error. This can happen cause of misconfigured API gateway or reverse proxy. It can also be caused by a misconfiguration in the CORS origin field. 
+
+We can remedy this issue by ensuring the origin field lines up with the API domain located in other parts of the code. I’ve highlighted them in bold below -
+
+```tsx
+import supertokens from "supertokens-node";
+import express from "express";
+import cors from "cors";
+
+supertokens.init({
+  supertokens: {
+    connectionURI: "...",
+  },
+
+  appInfo: {
+    appName: "...",
+    apiDomain: "http://localhost:3001", 
+    websiteDomain: "http://localhost:3000",
+  },
+   recipeList: [/* ... */],
+});
+
+const app = express();
+app.use(
+   cors({
+      //highlight-start
+      origin: "http://localhost:3000", // Check this line
+      //highlight-end
+      allowedHeaders: ["content-type", ...supertokens.getAllCORSHeaders()],
+      methods: ["GET", "PUT", "POST", "DELETE"],
+      credentials: true,
+   })
+);
+```
+
+<br />
+
+### `[Error] Origin http://localhost:3000 is not allowed by Access-Control-Allow-Origin. Status code: 200`
+
+There are two cases where this might happen. 
+
+1. When the `Access-Control-Allow-Origin` doesn't have `http://localhost:3000` in its values. 
+2. The CORS middleware is after the Supertokens middleware - in this case, the OPTIONS API will have the right value for `Access-Control-Allow-Origin`, but the actual `GET` / `POST` / `PUT` API won't since the Supertokens middleware will return a response before the CORS middleware is even called. To fix this, simply put the CORS middleware before the Supertokens middleware.
+
+```tsx
+import supertokens from "supertokens-node";
+import express from "express";
+import cors from "cors";
+import middleware from "supertokens-node/framework/express";
+
+const app = express();
+
+app.use(
+   cors({
+       origin: "http://localhost:3000", // Make sure this is the correct domain
+       allowedHeaders: ["content-type", ...supertokens.getAllCORSHeaders()],
+       methods: ["GET", "PUT", "POST", "DELETE"],
+       credentials: true,
+   })
+);
+//highlight-start
+app.use(middleware()); // CORS should be before this line
+//highlight-end
+```
+
+<br />
+
+### `[Error] Cannot use wildcard in Access-Control-Allow-Origin when credentials flag is true.`
+
+This appears when `Access-Control-Allow-Origin` header is set to `*`. To fix, add in the correct origin value in the CORS settings.
+
+```tsx
+import supertokens from "supertokens-node";
+import express from "express";
+import cors from "cors";
+
+const app = express();
+
+app.use(
+   cors({
+      //highlight-start
+      origin: "http://localhost:3000",  // Make sure the website domain is correct
```

Comment:
```suggestion
      origin: "http://localhost:3000",  // Make sure this is the same we the website domain
```

Reply:
Fixed, and also changed other parts of the doc where code comments might be confusing.